File: ruby/src/azure/mgmt/microsoft_compute/2016_06_30/models/disk.rb
require_relative "../../models/disk"


module Azure
  module Mgmt
    module Compute
      module Api_2016_06_30
        module Models
          class Disk < Azure::Mgmt::Compute::Models::Disk
            attr_accessor :disk_size_gb, :owner_id

            def initialize(name:, location:, tags: nil, disk_size_gb: nil, owner_id: nil)
              super(name: name, location: location, tags: tags)
              self.disk_size_gb = disk_size_gb
              self.owner_id = owner_id
            end
          end
        end
      end
    end
  end
end

File: dotnet/src/Azure/Microsoft.Azure.Compute._2016_06_30/Disk.cs
using System.Collections.Generic;

namespace Microsoft.Azure.Compute
{
    public abstract partial class _2016_06_30
    {
        public class Disk : Resource
        {
            public Disk(
                string id,
                string name,
                string type,
                string location,
                IDictionary<string, string> tags = null,
                int? diskSizeGB = null,
                string ownerId = null) : base(id, name, type, location, tags)
            {
                DiskSizeGB = diskSizeGB;
                OwnerId = ownerId;
            }

            /// <summary>
            /// Gets or sets if creationData.createOption is Empty, this field is
            /// mandatory and it indicates the size of the VHD to create. If this
            /// field is present for updates or creation with other options, it
            /// indicates a resize. Resizes are only allowed if the disk is not
            /// attached to a running VM, and can only increase the disk's size.
            /// </summary>
            public int? DiskSizeGB { get; set; }

            /// <summary>
            /// Gets a relative URI containing the VM id that has the disk
            /// attached.
            /// </summary>
            public string OwnerId { get; set; }

            public string ProvisioningState { get; }
        }
    }
}
